apps/nextjs-website/src/components/organisms/Auth/ConfirmLogin.tsx

```diff
@@ -28,10 +28,16 @@ const ConfirmLogin = ({ onBackStep, onConfirmLogin }: confirmLoginProps) => {
   } = translations;
 
   const [error, setError] = useState<string | null>(null);
+  const [submitting, setSubmitting] = useState(false);
   const [code, setCode] = useState<string>('');
 
-  const onconfirmLoginHandler = useCallback(() => {
-    onConfirmLogin(code).catch((e) => setError(e.message));
+  const onConfirmLoginHandler = useCallback(() => {
+    setSubmitting(true);
+    onConfirmLogin(code)
+      .catch((e) => setError(e.message))
+      .finally(() => {
+        setSubmitting(false);
+      });
```

Comment:
If you set `submitting` to `true` there is a time - when the user has been loaded, but the redirect to the home page has not yet been completed - in which you can click the submit button again. 

```suggestion
    onConfirmLogin(code).catch((e) => setError(e.message));
```

Reply:
I removed the finally step and moved the `setSubmitting(false)` call inside the catch step because the only case in which we need to reset the submission state is when the onConfirm callback throws an error otherwise, on success, the user will be redirected to homepage.